### PR TITLE
Fix opset5::Loop specification for sliced input

### PR DESCRIPTION
**Pull request by itikhono**
tickets 43626, 43620
- Fixed opset5::Loop spec: the sliced input is specified
- Aligned opset5::Loop implementation and reference implementation to the spec
- Fixed reshaping in validate_and_infer_type function of ngraph impl of Loop op
- Implemented new single layer tests to cover the cases (by @AlexPeskov )
- Shape inference unit test to cover sliced input of Loop
- Reshaping tests for Loop and TensorIterator

### CODE REVIEW

**Review by lazarevevgeny — APPROVED** [review]
Spec looks good.

File: inference-engine/cmake/features_ie.cmake [hunk 1]
```diff
@@ -98,6 +98,8 @@ ie_option (ENABLE_PYTHON "enables ie python bridge build" OFF)
 
 ie_option (ENABLE_V7_SERIALIZE "enables serialization to IR v7" OFF)
 
+ie_option (ENABLE_V10_SERIALIZE "enables experimental serialization to IR v10" OFF)
```
Comment (ilyachur) [inline inference-engine/cmake/features_ie.cmake:101]: Why do you need this? V10 serializer is enabled for default in the master branch.
We don't need to cherry-pick this commit from the release branch.

**Review by ilyachur — COMMENTED** [review]
Looks like it is a mix of different commits from the release branch + Loop.

We shouldn't have a commit 5abbe2fec553745e066e29950f4b09663fec16b8 in the master branch.

File: inference-engine/cmake/features_ie.cmake (hunk 1 above)
Comment (itikhono) [reply inference-engine/cmake/features_ie.cmake]: reverted

File: ngraph/core/src/op/loop.cpp [hunk 2]
```diff
@@ -182,34 +180,38 @@ void op::v5::Loop::validate_and_infer_types()
...
+                // infer type for m_body_parameter
+                Shape out_shape{input_partial_shape.to_shape()};
+                out_shape[slice_input_description->m_axis] = slice_input_description->m_part_size;
+                body_parameter->set_partial_shape(out_shape);
+            }
+            else
+            {
+                body_parameter->set_partial_shape(
+                    PartialShape::dynamic(input_partial_shape.rank()));
+            }
```
Comment (jane-intel) [inline ngraph/core/src/op/loop.cpp:199]: nit:
```suggestion
            if (input_partial_shape.rank().is_static())
            {
                // infer type for m_body_parameter
                PartialShape out_shape{input_partial_shape};
                out_shape[slice_input_description->m_axis] = 
                        Dimension(slice_input_description->m_part_size);
                body_parameter->set_partial_shape(out_shape);
            }
            else
            {
                body_parameter->set_partial_shape(PartialShape::dynamic());
            }
```

**Review by GlebKazantaev — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Review by jane-intel — APPROVED** [review]

**Review by AlexPeskov — APPROVED** [review]
LGTM

File: ngraph/core/src/op/loop.cpp (hunk 2 above)
Comment (itikhono) [reply ngraph/core/src/op/loop.cpp:199]: it will be fixed in the next PR.

**Comment by jane-intel** [comment]
Approved by release committee.

**Review by jane-intel — APPROVED** [review]

**Pull request merged**